### ISSUE TITLE
Can fairseq be used on other problems?

### ISSUE THREAD

**Issue by PratsBhatt** [facebookresearch/fairseq-lua#15]
I just read about it and I am interested in using it. I am curious to know whether it can be used on other seq2seq problems which a RNN (e.g LSTM, GRU cells) are able to solve?
E.g Image captioning, Document summarization.
Thank you and amazing work!
Lookin forward for your positive response.
Regards,

**Comment by jgehring** [comment]
So far, we have evaluated the model on translation and (sentence) summarization. The image captioning corpora that I'm aware of are usually too small to do end-to-end training. Anyway, regarding image-to-text: it should be possible to load 2-dimensional tensors from a `tnt.IndexedDataset` with relatively minor modifications (mostly related to batching) and to change the encoder part of the model to work on image instead of text data.

tl;dr: out-of-the box, we only support 1D-to-1D modeling (seq2seq) and supporting other formats requires modifying the code (but no major refactoring).

**Comment by SuperWu090** [comment]
@PratsBhatt Have you successfully used the fairseq in 2D data (like image) problem? I am also working on it.

**Comment by PratsBhatt** [comment]
@SuperWu090  No I have not, I am trying to first get my hands dirty with the algo so I am currently trying out with other text data.

**Comment by harsh-agarwal** [comment]
Well, I am trying to something inline to that though in caffe. 

However, as I am new to torch I was referring to [convseq2seq](https://github.com/tobyyouup/conv_seq2seq) which implements this model in tensorflow.

I am facing troubles implementing a decoder for images as input.

As far as I understood the output token that we find is concatenated to the input of the decoder till you reach the end of the sentence. But how has that been implemented? Is it something like an RNN or is it like running multiple forward passes each time changing my input to the decoder. Even for sentences? 

Also, I would be grateful if someone could verify the encoder that I have made based on my understanding of the code. The following is the link to visualize the network that I have made. 

[caffe_conv_encoder_visualisation](http://ethereon.github.io/netscope/#/gist/cc5ff9df4cdcdbbd2516b320dbed582f)

Ignore the AlexNet part that is there as I am trying to find a sequence in an image

The network after pool5 is my encoder.

Struggling hard since a long time on this. Any help would be appreciated :)

Cheers
Harsh Agarwal
